pkg/client/cr.go

```diff
@@ -15,12 +15,56 @@
 package client
 
 import (
+	"context"
+
 	"github.com/coreos/etcd-operator/pkg/spec"
+	"github.com/coreos/etcd-operator/pkg/util/k8sutil"
+
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/runtime/serializer"
 	"k8s.io/client-go/rest"
 )
 
+type EtcdClusterCR interface {
+	// Create creates an etcd cluster CR with the desired CR
+	Create(ctx context.Context, namespace string, cl *spec.EtcdCluster) (*spec.EtcdCluster, error)
+
+	// Get returns the specified etcd cluster CR
+	Get(ctx context.Context, name, namespace string) (*spec.EtcdCluster, error)
+
+	// Delete deletes the specified etcd cluster CR
+	Delete(ctx context.Context, name, namespace string) error
+
+	// Update updates the etcd cluster CR.
+	Update(ctx context.Context, name, namespace string, etcdCluster *spec.EtcdCluster) (*spec.EtcdCluster, error)
+}
+
+type etcdClusterCR struct {
+	client     *rest.RESTClient
+	crScheme   *runtime.Scheme
+	paramCodec runtime.ParameterCodec
+}
+
+func MustNewInCluster() (EtcdClusterCR, error) {
+	cfg, err := k8sutil.InClusterConfig()
+	if err != nil {
+		return nil, err
+	}
+	return NewEtcdClusterCR(cfg)
+}
+
+func NewEtcdClusterCR(cfg *rest.Config) (EtcdClusterCR, error) {
+	cli, crScheme, err := New(cfg)
+	if err != nil {
+		return nil, err
+	}
+	return &etcdClusterCR{
+		client:     cli,
+		crScheme:   crScheme,
+		paramCodec: runtime.NewParameterCodec(crScheme),
+	}, nil
+}
+
 func New(cfg *rest.Config) (*rest.RESTClient, *runtime.Scheme, error) {
```

Comment:
add a TODO: `make this private so that we don't expose RESTClient once operator code uses this client instead of REST calls`